src/soc/starfive/jh7100/src/clock.rs

```diff
@@ -256,7 +256,7 @@ pub fn _ENABLE_CLOCK_clk_cpundbus_root_() {}
 pub fn _SWITCH_CLOCK_clk_cpundbus_root_SOURCE_clk_osc_sys_() {
     let mut v = peek32(clk_cpundbus_root_ctrl_REG_ADDR);
     v &= !(0x3 << 24);
-    v |= (0x0 & 0x3) << 24;
+    v |= 0 << 24;
```

Comment:
I know you're just satisfying clippy, so maybe it's out of scope, but in many of these places, there's further simplication: 0 << 24 == 0. v | 0 == v, so this line can be removed entirely. It also feels like we're missing an opportunity to use a Rust zero-cost abstraction here for safety. Perhaps there's a good bitflags crate to use here to reduce opaque magic numbers everywhere.

Reply:
I agree and those changes are coming.

Reply:
This was the output of a script pipeline from C. The C itself was a crime against nature. So it's a bit of a mess, but also way too much code to do by hand, and I"m not smart enough to do it with anything but sed.